manage_breast_screening/notifications/management/commands/helpers/message_batch_helpers.py

```diff
@@ -61,10 +62,45 @@ def mark_batch_as_failed(
                     }
                 )
             )
+        elif response.status_code == VALIDATION_ERROR_STATUS_CODE:
+            MessageBatchHelpers.process_validation_errors(message_batch, retry_count)
         else:
             message_batch.status = MessageBatchStatusChoices.FAILED_UNRECOVERABLE.value
         message_batch.save()
 
         for message in message_batch.messages.all():
             message.status = MessageStatusChoices.FAILED.value
             message.save()
+
+    @staticmethod
+    def process_validation_errors(message_batch: MessageBatch, retry_count: int = 0):
+        message_batch_errors = json.loads(message_batch.nhs_notify_errors).get("errors")
+
+        new_message_batch = MessageBatch.objects.create(
```

Comment:
Would it be cleaner from an audit trail perspective to preserve the message batch in error (the original) along with the nhs notify errors and the messages in error. Mark it as failed unrecoverable. Create a new message batch, move the valid messages to it and send the new batch?
Asking because the original failed batch is useful to us from a support perspective along with any timestamps around the failure. I don't think we should be updating it.
One other point here is that the new message batch is created regardless of the success of the regex capture, if something goes wrong with the regex we have an orphan failed batch record.
A bit to unpick here so happy to chat about it.